src/coreclr/pal/src/map/virtual.cpp

```diff
@@ -2184,9 +2196,39 @@ void ExecutableMemoryAllocator::TryReserveInitialMemory()
         preferredStartAddress += preferredStartAddressIncrement;
 
     } while (sizeOfAllocation >= MemoryProbingIncrement);
+#else
+
+    // Always try to allocate above the location of libcoreclr on arm - we want to start allocating near it (withih 128Mb distance)
```

Comment:
I have noticed that the available memory was above the coreclr on arm64 too. However, I wonder if it is safe to assume that it will always be the case. Maybe we should search in both locations.

Reply:
From my understanding if it's below than we will unlikely to benefit from it as is: 
![image](https://user-images.githubusercontent.com/523221/173599182-25e5cd19-6501-48fd-a1a0-33117955a087.png)

we'll have to modify allocator to do right->left allocation, am I right?

Reply:
Added a quick path for "below users space" case